nautobot/utilities/drf_views.py

```diff
@@ -0,0 +1,815 @@
+        queryset lookups.  Eg if objects are referenced using multiple
+        keyword arguments in the url conf.
+        """
+        queryset = self.filter_queryset(self.get_queryset())
+
+        # Perform the lookup filtering.
+        lookup_url_kwarg = self.lookup_url_kwarg or self.lookup_field
+        if lookup_url_kwarg not in self.kwargs:
+            return self.queryset.model()
+        filter_kwargs = {self.lookup_field: self.kwargs[lookup_url_kwarg]}
+        obj = get_object_or_404(queryset, **filter_kwargs)
+
+        # May raise a permission denied
+        self.check_object_permissions(self.request, obj)
+
+        return obj
+
+    def get_extra_context(self, request, view_type, instance=None):
+        """
+        Return any additional context data for the template.
+        request: The current request
+        instance: The object being viewed
+        """
+        return {}
+
+    def get_template_name(self, view_type):
+        # Use "<app>/<model>_<view_type> if available, else fall back to generic templates
+        model_opts = self.model._meta
+        app_label = model_opts.app_label
+        if view_type == "detail":
+            return f"{app_label}/{model_opts.model_name}.html"
+
+        try:
+            select_template([f"{app_label}/{model_opts.model_name}_{view_type}.html"])
+            return f"{app_label}/{model_opts.model_name}_{view_type}.html"
+        except TemplateDoesNotExist:
+            return f"generic/object_{view_type}.html"
+
+    def retrieve_object_bulk(self, request, form, view_type, **kwargs):
+        pk_list = kwargs.pop("pk_list")
+        table = self.table(self.queryset.filter(pk__in=pk_list), orderable=False)
+        if not table.rows:
+            messages.warning(
+                request,
+                f"No {self.queryset.model._meta.verbose_name_plural} were selected for deletion.",
+            )
+            return redirect(self.get_return_url(request))
+
+        data = {
+            "form": form,
+            "table": table,
+            "obj_type_plural": self.queryset.model._meta.verbose_name_plural,
+            "return_url": self.get_return_url(request),
+            "template": self.get_template_name(view_type),
+        }
+        data.update(self.get_extra_context(request, view_type, instance=None))
+        return Response(data)
+
+    def initial(self, request, *args, **kwargs):
+        """
+        Runs anything that needs to occur prior to calling the method handler.
+        """
+        self.format_kwarg = self.get_format_suffix(**kwargs)
+
+        # Perform content negotiation and store the accepted info on the request
+        neg = self.perform_content_negotiation(request)
+        request.accepted_renderer, request.accepted_media_type = neg
+
+        # Determine the API version, if versioning is in use.
+        version, scheme = self.determine_version(request, *args, **kwargs)
+        request.version, request.versioning_scheme = version, scheme
```

Comment:
These are REST API specific, aren't they? Not sure how they apply to UI views.

Reply:
We are using BrowableAPIRenderer to render the forms and templates. I have to specify `accepted_renderer`, and `accepted_media_type` beforehand or else it will throw an error.
<img width="1618" alt="Screen Shot 2022-08-01 at 8 39 44 PM" src="https://user-images.githubusercontent.com/46973263/182267676-ff2bdaf8-6a1d-46d1-ab65-c9f40a0bff87.png">